### DISCUSSION TITLE
How to see git diff for `.db` file?

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#3651] [Q&A]
Hi. I have this in vsCode git diff explorer, How can I open this `.db` file and see the differences like the vsCode diff editor?

![image](https://github.com/pocketbase/pocketbase/assets/44144724/bdd29b04-d07e-4f84-a6e9-8dcdecd4b675)

Like this (vsCode diff editor):

![image](https://github.com/pocketbase/pocketbase/assets/44144724/750a1863-5c6f-471f-a6f1-aa1ae283f0df)

Is there a vsCode extension for this?

**Comment by ganigeorgiev** [comment]
The `pb_data` directory and the database files inside it usually should NOT be committed (unless for example you are using it for "fixtures", aka. to run your Go tests on it).

I don't use VSCode and I'm not sure if there is such extension but you can search for "SQLite git diff" or similar (_but even if exists I don't think this can be done efficiently for large databases_).

**Comment by stephdin** [comment]
There is a way to configure git to recognize sqlite database files and track the changes. This blogpost describes the process quite well: https://garrit.xyz/posts/2023-11-01-tracking-sqlite-database-changes-in-git
